Essentials/src/com/earth2me/essentials/User.java

```diff
@@ -599,6 +599,26 @@ public boolean checkMuteTimeout(final long currentTime) {
         return false;
     }
 
+    //Returns a list of UUIDs who's ignore timeout expired during this check
+    public List<UUID> checkIgnoreTimeouts(final long currentTime)
+    {
+        List<UUID> expiredIgnoredPlayers = new ArrayList();
+        if (ignoredPlayerTimeouts != null) {
+            Iterator ignoredPlayers = ignoredPlayerTimeouts.entrySet().iterator();
+            while (ignoredPlayers.hasNext())
```

Comment:
Absolutely no reason to use an Iterator. Should be using an enhanced for loop.

Reply:
Using the new timer checking method, there is no longer iteration over the timeouts.